src/libsystemd/meson.build

```diff
@@ -232,7 +232,8 @@ tests += [
           libglib,
           libgobject,
           libgio,
-          libdbus]],
+          libdbus],
```

Comment:
I think you want to use a explicit `if libdbus.found()` check for this code like in #21895 to ensure the disabler doesn't propagate too far and disable all the tests here.

Reply:
If libdbus is not found, then HAVE_DBUS is 0. And we usually control enable or disable by the condition HAVE_XXX.
Or you mean this doesn't work?

Reply:
> If libdbus is not found, then HAVE_DBUS is 0. And we usually control enable or disable by the condition HAVE_XXX.

Different issue, [disablers](https://mesonbuild.com/Disabler.html) depending on how they are adding to lists will generally disable the entire list and anything that depends on that list I think. By making the list addition for `libdbus` conditional on `libdbus.found()` which is a boolean we can guard against unintended behavior.

Reply:
> > If libdbus is not found, then HAVE_DBUS is 0. And we usually control enable or disable by the condition HAVE_XXX.
> 
> Different issue, [disablers](https://mesonbuild.com/Disabler.html) depending on how they are adding to lists will generally disable the entire list and anything that depends on that list I think. By making the list addition for `libdbus` conditional on `libdbus.found()` which is a boolean we can guard against unintended behavior.

Sorry, I cannot follow... What problem did you see?

I noticed that setting `HAVE_DBUS` in the condition is even not necessary, as now libdbus is a dependency or disabler object. So, in the current version, nothing changed for the test-bus-marshal.

Reply:
Hm, I noticed that test-bus-marshal can be built without libdbus. Using disabler is also dropped.